firestore/objc/firestore-smoketest-objc/ViewController.m

```diff
@@ -593,6 +602,42 @@ - (void)exampleData {
   // [END example_data]
 }
 
+- (void)exampleDataCollectionGroup {
+    // [START fs_collection_group_query_data_setup]
+    FIRCollectionReference *citiesRef = [self.db collectionWithPath:@"cities"];
+
+    NSDictionary *data = @{@"name": @"Golden Gate Bridge", @"type": @"bridge"};
+    [[[citiesRef documentWithPath:@"SF"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Legion of Honor", @"type": @"museum"};
+    [[[citiesRef documentWithPath:@"SF"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Griffith Park", @"type": @"park"};
+    [[[citiesRef documentWithPath:@"LA"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"The Getty", @"type": @"museum"};
+    [[[citiesRef documentWithPath:@"LA"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Lincoln Memorial", @"type": @"memorial"};
+    [[[citiesRef documentWithPath:@"DC"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"National Air and Space Museum", @"type": @"museum"};
+    [[[citiesRef documentWithPath:@"DC"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Ueno Park", @"type": @"park"};
+    [[[citiesRef documentWithPath:@"TOK"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"National Museum of Nature and Science", @"type": @"museum"};
+    [[[citiesRef documentWithPath:@"TOK"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Jingshan Park", @"type": @"park"};
+    [[[citiesRef documentWithPath:@"BJ"] collectionWithPath:@"landmarks"] addDocumentWithData:data];
+
+    data = @{@"name": @"Beijing Ancient Observatory", @"type": @"museum"};
```

Comment:
Is it safe to keep using `data` as the name?  I assume each call is by value and not by reference?

Reply:
`data` is an object pointer that's pointed to a new reference type in each line before the `addDocumentWithData` call is invoked. It's lazy coding, but makes the already cumbersome objc chained calls slightly less cumbersome.